epam_tr/src/main/java/com/epam/tr/service/UserServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.tr.service;
+
+import com.epam.tr.dao.Dao;
+import com.epam.tr.entities.AppUser;
+import com.epam.tr.entities.UserRole;
+import com.epam.tr.exceptions.InvalidCredentialsException;
+import com.epam.tr.service.logic.validator.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private Dao<AppUser> dao;
+    private Validator<AppUser> validator;
+
+    @Autowired
+    public UserServiceImpl(Dao<AppUser> dao, Validator<AppUser> validator) {
+        this.dao = dao;
+        this.validator = validator;
+    }
+
+    @Override
+    public void create(AppUser entity) throws InvalidCredentialsException {
+        checkValid(entity);
+        dao.insert(entity);
+    }
+
+    private void checkValid(AppUser entity) throws InvalidCredentialsException {
+        if (!validator.isValid(entity)) {
+            throw new InvalidCredentialsException("Invalid login or password");
+        }
+    }
+
+    @Override
+    public void update(AppUser oldEntity, AppUser newEntity) throws InvalidCredentialsException {
+        checkValid(oldEntity);
```

Comment:
why do you validate oldEntity if you retrive it from DB?
plus I think you should retrive it from DB here, not in a controller - same for the rest.